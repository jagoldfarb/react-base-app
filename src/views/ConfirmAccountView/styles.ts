import styled from 'styled-components';

import { COLORS } from 'helpers/enums/colors';

export const ConfirmAccountContainer = styled.div`
  background-color: ${COLORS.white};
`;
