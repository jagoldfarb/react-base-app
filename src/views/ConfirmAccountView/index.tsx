import React, { FunctionComponent } from 'react';
import i18next from 'i18next';

import Loading from 'components/commons/Loading';

import { ConfirmAccountContainer } from './styles';
import './i18n';

const ConfirmAccountView: FunctionComponent = () => (
  <ConfirmAccountContainer data-testid="confirm-account-view">
    <Loading message={i18next.t('CONFIRM_ACCOUNT_VIEW:MESSAGE')} />
  </ConfirmAccountContainer>
);

export default ConfirmAccountView;
