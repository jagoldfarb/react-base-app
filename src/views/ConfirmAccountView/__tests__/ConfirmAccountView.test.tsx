import React from 'react';
import { render } from '@testing-library/react';

import ConfirmAccountView from '..';

describe('ConfirmAccountView view', () => {
  it('Should render the confirm account page', () => {
    const { asFragment, getByTestId } = render(<ConfirmAccountView />);
    expect(getByTestId('confirm-account-view')).toBeDefined();
    expect(getByTestId('loader')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
