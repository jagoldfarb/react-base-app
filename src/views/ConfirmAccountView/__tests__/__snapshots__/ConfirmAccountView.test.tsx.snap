// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmAccountView view Should render the confirm account page 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnbm jxbMaU"
    data-testid="confirm-account-view"
  >
    <div
      class="sc-bdfBwQ cyXInH"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        data-testid="loader"
        role="progressbar"
        style="width: 65px; height: 65px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
      <span
        class="sc-gsTCUz ISRsW"
      >
        The account is being verified
      </span>
    </div>
  </div>
</DocumentFragment>
`;
